dm/pkg/storage/utils.go

```diff
@@ -40,21 +41,43 @@ func AdjustPath(rawURL string, uniqueID string) (string, error) {
 	}
 	// not url format, we don't use url library to avoid being escaped or unescaped
 	if u.Scheme == "" {
-		// avoid duplicate add uniqueID
-		if !strings.HasSuffix(rawURL, uniqueID) {
-			return rawURL + uniqueID, nil
+		// avoid duplicate add uniqueID, and trim suffix '/' like './dump_data/'
+		trimPath := strings.TrimRight(rawURL, string(filepath.Separator))
+		if !strings.HasSuffix(trimPath, uniqueID) {
+			return trimPath + uniqueID, nil
 		}
 		return rawURL, nil
 	}
 	// u.Path is an unescaped string and can be used as normal
-	if !strings.HasSuffix(u.Path, uniqueID) {
-		u.Path += uniqueID
+	trimPath := strings.TrimRight(u.Path, string(filepath.Separator))
+	if !strings.HasSuffix(trimPath, uniqueID) {
+		u.Path = trimPath + uniqueID
 		// u.String will return escaped url and can be used safely in other steps
 		return u.String(), err
 	}
 	return rawURL, nil
 }
 
+// TrimPath trims rawURL suffix which is uniqueID, supports local and s3.
+func TrimPath(rawURL string, uniqueID string) (string, error) {
+	if rawURL == "" || uniqueID == "" {
+		return rawURL, nil
+	}
+	u, err := bstorage.ParseRawURL(rawURL)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	// not url format, we don't use url library to avoid being escaped or unescaped
+	if u.Scheme == "" {
+		// avoid duplicate add uniqueID, and trim suffix '/' like './dump_data/'
```

Comment:
Seems this function is only used for the subtask config, and subtask config will not have `/` suffix? we can remove this comment or in later PR

Reply:
This is for handle the last `/` of user configuration like `./dump_data/`.  this issue was found out by @sunzhaoyang when discuss how to add subdir.

Reply:
ok, i will fix this comment in later pr.